azurerm/helpers/validate/domain_name.go

```diff
@@ -0,0 +1,23 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// Fqdn validates that a domain name, including the host portion, is valid to RFC requirements
+// e.g. portal.azure.com
+func DomainName(i interface{}, k string) (warnings []string, errors []error) {
+	//
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected %q to be string", k))
+		return
+	}
+	// The following regexp is a good example of why pre-processing support would be nice in GoLang
+	if matched := regexp.MustCompile(`^(?:[_a-z0-9](?:[_a-z0-9-]{0,61}[a-z0-9]\.)|(?:[0-9]+/[0-9]{2})\.)+(?:[a-z](?:[a-z0-9-]{0,61}[a-z0-9])?)?$`).Match([]byte(v)); !matched || len(v) > 253 {
```

Comment:
could we add some tests for this? I have a feeling we may be able to simplify this to:

```
if _, err := uri.Parse(fmt.Sprintf("http://%s/", v)); err != nil {
  ...
}
```

which should handle this for us?

Reply:
As discussed offline, that will unfortunately accept any string that meets allowed character requirements, rather than qualifying complete hostname.